### PR TITLE
Add "getDefinitionAtPosition" to interface

### PR DESCRIPTION
**Pull request by divyenduz**
This would help implement "go to definition" feature within a template literal.

### CODE REVIEW

**Comment by mjbvz** [comment]
Makes sense. Can you please add the implementation and a quick test for this new api as well

**Comment by divyenduz** [comment]
Sure! I will do that but tests are failing for me on local and in CI, even for master branch.

**Comment by mjbvz** [comment]
Use node 7. Node 8+ broke something for the tests and I haven't had time to look into it yet

**Comment by mjbvz** [comment]
Tests should run on node 8+ now too. Just make sure to merge in the latest master

**Comment by divyenduz** [comment]
@mjbvz : I attempted implementation of `getDefinitionAtPosition` and related test case. I have the following concerns:-

1. Lint fails for lines that are not touched by this PR.
1. Generally, how do we know the [command string](https://github.com/Microsoft/typescript-template-language-service-decorator/pull/7/files#diff-0af839787ac460606c9fcf1dcb961b38R47) for a language feature? This is not from LSP. I just took a guess by looking at [TypeScript service](https://github.com/Microsoft/TypeScript/tree/5f4a03c40875e1e442ae6168df8c322d22496a1b/src/services) source code.
1. How do we know the functions that can be added to the interface. Again, my approach was to look at the [TS service source](https://github.com/Microsoft/TypeScript/blob/5f4a03c40875e1e442ae6168df8c322d22496a1b/src/services/goToDefinition.ts#L3). 
1. I have a concern in [one of the test cases](https://github.com/Microsoft/typescript-template-language-service-decorator/pull/7/files#diff-0af839787ac460606c9fcf1dcb961b38R37) and I can use your feedback. 

Thanks!

**Comment by mjbvz** [comment]
Thanks for the followup. To your points:

1. I think the code has a syntax error. Let me know if this is not the case

2. Those are TS server commands. They are documented [here](https://github.com/Microsoft/TypeScript/blob/master/lib/protocol.d.ts)

3. This library overrides methods in the typescript language service. Features like code lenses are specific to VS Code but VS Code does use the typescript language apis to power code lenses

4. I think the offset in the test may be incorrect. Try getting rid of the leading whitespace to make the real offset clearer

5. I don't understand this question. Can you please provide an example of what you are trying to do?

**Comment by divyenduz** [comment]
Thanks for the detailed answers, really appreciate it. I will fix the PR tomorrow. 

5. I want to test a sample implementation of `getDefinitionAtPosition` which will usually be provided when using this library in another extension? Is it possible to test such a scenario.

File: e2e/tests/getDefinitionAtPosition/node_modules/echo-plugin/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// @ts-check
+const createPlugin = require('../../../../_plugin');
+
+/**
+ * @augments {TemplateStringLanguageService}
+ */
+class TestStringLanguageService {
```
Comment (mjbvz) [inline e2e/tests/getDefinitionAtPosition/node_modules/echo-plugin/index.js]: For 5, update this class to implement `getDefinitionAtPosition` instead. Make it return some fake definition position that still verifies that the implementation was called with the right arguments

**Comment by divyenduz** [comment]
Thanks for the feedback. 

1. Looks like it was a syntax issue indeed that I missed (and prettier). It is resolved now. 
4. I tried to fix the test case by removing leading whitespace but no luck.

File: e2e/tests/getDefinitionAtPosition/test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            // Default goToDefinition inside tagged template literal should be blank
+            assert.strictEqual(definitions.length, 0);
+        })
+    });
+
+    it('should return definition at position', async () => {
+        return getDefinitionAtPositionInMockFile(
+            `const a = 'test';
+console.log(a);`,
+            { offset: 12, line: 1 },
```
Comment (mjbvz) [inline e2e/tests/getDefinitionAtPosition/test.js]: Line and column start at one, not zero. Try: line 2, offset 13 (or maybe 14)

Comment (divyenduz) [reply e2e/tests/getDefinitionAtPosition/test.js]: I tried the suggested values already, but no luck.

Comment (divyenduz) [reply e2e/tests/getDefinitionAtPosition/test.js]: Let me make the variable multi-character and choose an index in the middle.

Comment (divyenduz) [reply e2e/tests/getDefinitionAtPosition/test.js]: Nope, no luck even with that, anyways pushing it as it is less ambiguous!

**Comment by mjbvz** [comment]
I'm merging this as-is and will try fixing the tests. Thanks for the addition @divyenduz!

**Pull request merged**